extensions/amp-access/0.1/amp-access.js

```diff
@@ -770,17 +777,43 @@ export class AccessService {
   }
 
   /**
+   * Runs the login flow using one of the predefined urls in the amp-access config
+   *
+   * @param {string} type Type of login defined in the config
+   * @return {!Promise}
+   */
+  loginWithType_(type) {
+    // Login URL should always be available at this time.
+    const loginUrl = user().assert(this.loginUrlMap_[type],
+        'Login URL is not ready: %s', type);
+    user().assert(this.loginConfig_[type],
```

Comment:
Could you change the order? The `loginConfig_` assert should go before `loginUrlMap_`: the record absent in the `loginConfig_` is a misconfiguration and it will always mean that `loginUrlMap_` is also missing. The reverse is not true, however.